Module01/Lexer.py

```diff
@@ -1,288 +1 @@
-            self.is_number = False
-            return ('VALUE', self.current_token)
-        return ('EOF', None)
-
-def runlexer(file_in):
-    lexer = Lexer(file_in)              # 1. Initialize the lexer with the input string
-    token = lexer.get_next_token()      # 2. Get the first token from the lexer
-    tokens = ""                         # For testing get output in string instead of print line
-    while token[0] != 'EOF':            # 3. Loop until the end of the input is reached
-        tokens += "["+token[0]+"]"+"["+token[1]+"]"                    # 4. save the current tokens
-        token = lexer.get_next_token()  # 5. Get the next token
-    return tokens                       # return results
-
-"""
-Test Cases
-Includes various arithmetic expressions to test lexer and correctly identify and tokenize each component
-"""
-def testalpha():
-    testfile1 = runlexer("betrius = 2 + 3 / 4 * 5") # multiple letters and multiple math signs
-    successes = 0
-    if(testfile1 == "[VARIABLE][betrius][SPACE][ ][ASSIGN][=][SPACE][ ][VALUE][2][SPACE][ ][ADD][+][SPACE][ ][VALUE][3][SPACE][ ][DIVIDE][/][SPACE][ ][VALUE][4][SPACE][ ][MULTIPLY][*][SPACE][ ][VALUE][5]"):
-        successes +=1
-
-    #testfile2 = runlexer(" 2letrius = 2 + 3 / 4 * 5") # Starting with a number prior to =
-    #successes = 0
-    #if(testfile2 == ""):
-    #    successes +=1
-    #This test should: raise Exception("Cannot have a number value prior to assignment operator.")
-
-    testfile3 = runlexer(" celcius = 257 + 2 ") # starting and ending with spaces
-    print(testfile3)
-    if(testfile3 == "[SPACE][ ][VARIABLE][celcius][SPACE][ ][ASSIGN][=][SPACE][ ][VALUE][257][SPACE][ ][ADD][+][SPACE][ ][VALUE][2][SPACE][ ]"):
-        successes +=1
-    
-    testfile4 = runlexer("a=1+1") #testing add, no spaces
-    print(testfile4)
-    if(testfile4 == "[VARIABLE][a][ASSIGN][=][VALUE][1][ADD][+][VALUE][1]"):
-        successes +=1
-
-    testfile5 = runlexer("b=2-1") #testing subtract, no spaces
-    print(testfile5)
-    if(testfile5 == "[VARIABLE][b][ASSIGN][=][VALUE][2][SUBTRACT][-][VALUE][1]"):
-        successes +=1
-
-    testfile6 = runlexer("c=1*1") #testing multiply, no spaces
-    print(testfile6)
-    if(testfile6 == "[VARIABLE][c][ASSIGN][=][VALUE][1][MULTIPLY][*][VALUE][1]"):
-        successes +=1
-
-    testfile7 = runlexer("d=4/2") #testing divide, no spaces
-    print(testfile7)
-    if(testfile7 == "[VARIABLE][d][ASSIGN][=][VALUE][4][DIVIDE][/][VALUE][2]"):
-        successes +=1
-
-    testfile8 = runlexer("e=10000000000/1000") #testing large values, no spaces
-    print(testfile8)
-    if(testfile8 == "[VARIABLE][e][ASSIGN][=][VALUE][10000000000][DIVIDE][/][VALUE][1000]"):
-        successes +=1
-
-    testfile9 = runlexer("e=100000 00000/1000") #testing large values, with space in value
-    print(testfile9)
-    if(testfile9 == "[VARIABLE][e][ASSIGN][=][VALUE][100000][SPACE][ ][VALUE][00000][DIVIDE][/][VALUE][1000]"):
```

Comment:
If this is supposed to be valid, you may not want to tokenize space here. Tokenizing space could increase the complexity of your PEG.